src/main/java/mops/businesslogic/FileServiceImpl.java

```
@@ -0,0 +1,213 @@
+                .directory(dirId)
+                .owner(account.getName())
+                .tags(tags)
+                .build();
+
+        try {
+            FileInfo fileInfo = fileInfoService.saveFileInfo(meta);
+            fileRepository.saveFile(multipartFile, fileInfo.getId());
+        } catch (MopsException e) {
+            TransactionAspectSupport.currentTransactionStatus().setRollbackOnly();
+            log.error("Error while saving file {} by user {}. Error: {}",
+                    meta.getName(),
+                    account.getName(),
+                    e.getMessage(),
+                    new MopsException("Error while saving", e)
+            );
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    @SuppressWarnings({ "PMD.LawOfDemeter", "PMD.DataflowAnomalyAnalysis" })
+    @SuppressFBWarnings(value = "RCN_REDUNDANT_NULLCHECK_WOULD_HAVE_BEEN_A_NPE",
+            justification = "There's no nullcheck here.")
+    public FileContainer getFile(Account account, long fileId) throws MopsException {
+        FileInfo fileInfo;
+        try {
+            fileInfo = fileInfoService.fetchFileInfo(fileId);
+        } catch (MopsException e) {
+            log.error("User {} tried to retrieved non existing file with ID {}. Error: {}",
+                    account.getName(),
+                    fileId,
+                    e.getMessage()
+            );
+            // has to be extra thrown here, else there will be an initialization error
+            throw new FileNotFoundException("File with ID " + fileId + " not found", e);
+        }
+
+        UserPermission userPermission = directoryService.getPermissionsOfUser(account, fileInfo.getDirectoryId());
+
+        if (!userPermission.isRead()) {
+            log.error("User {} tried to read file {} without permission.",
+                    account.getName(),
+                    fileId,
+                    new ReadAccessPermissionException("No read permission")
```

you need to throw this exception instead of giving it to the logger.